### DISCUSSION TITLE
How to use vue.js templates in A3?

### DISCUSSION THREAD

**Discussion by andrewbenrichard** [apostrophecms/apostrophe#3260] [Q&A]
I have a project that is really complex and requires lots of logics, is it possible to create vue templates instead of .HTML files?

**Comment by boutell** [comment]
Hi Andrew,

You could use the REST APIs and build a headless application, in which case you're building a straight vue app that talks to your apostrophe "site" as a back end. You'd log into the site only as an admin interface.

Or you could build a hybrid application, in which you have a webpack build that pushes a Vue app to the `ui/public` folder of one of your modules, and that app handles the bits that don't feel good in Nunjucks. This can be good when Nunjucks is mostly right for your needs and Nunjucks does make SEO easy by generating server side pages.
